sklearn/cluster/_kmeans.py

```diff
@@ -93,12 +102,14 @@ def _k_init(X, n_clusters, x_squared_norms, random_state, n_local_trials=None):
         # that it helped.
         n_local_trials = 2 + int(np.log(n_clusters))
 
-    # Pick first center randomly
+    # Pick first center randomly and track index of point
     center_id = random_state.randint(n_samples)
+    indices = np.empty(n_clusters)
```

Comment:
```suggestion
    indices = np.empty(n_clusters, dtype=int)
```

Not sure yet, but it might be useful to initialize it with `np.full(-1, dtype=int)`. This way we can test that after running kmeans++, there's no -1 left.